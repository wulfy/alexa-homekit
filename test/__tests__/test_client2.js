const base_config = require("../config_tests/base_config");

//TEST FOR INDEX.JS
test('DISCOVERY TESTING client2', done => {

    const { DOMOTICZ_GET_DEVICES } = require("../mockups/client2Mockup");

    global.getDevices = (token,domoticzDeviceId) => {
        return JSON.parse(DOMOTICZ_GET_DEVICES).result;
    }
    let context2 = {};
    context2.succeed = function (data){
        const testData = JSON.stringify(data); 
        expect(testData).toMatchSnapshot();
        done();
    };
    
    base_config.handler(base_config.mockups.ALEXA_DISCOVERY_REQUEST_EXAMPLE,context2);

});