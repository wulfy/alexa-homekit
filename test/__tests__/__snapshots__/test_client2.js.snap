// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DISCOVERY TESTING client2 1`] = `"{\\"event\\":{\\"header\\":{\\"namespace\\":\\"Alexa.Discovery\\",\\"name\\":\\"Discover.Response\\",\\"payloadVersion\\":\\"3\\",\\"messageId\\":\\"abc-123-def-456\\"},\\"payload\\":{\\"endpoints\\":[{\\"endpointId\\":\\"1_undefined_undefined\\",\\"manufacturerName\\":\\"undefined\\",\\"friendlyName\\":\\"Volets Sud\\",\\"description\\":\\"\\",\\"displayCategories\\":[\\"SWITCH\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('On' === 'Off' || 'On' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"2_undefined_undefined\\",\\"manufacturerName\\":\\"undefined\\",\\"friendlyName\\":\\"Volets chambres\\",\\"description\\":\\"\\",\\"displayCategories\\":[\\"SWITCH\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('On' === 'Off' || 'On' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"1_YeeLightcoucoutuspciaux_Dimmer\\",\\"manufacturerName\\":\\"YeeLight coucou.tu  / àç!è§'( spéciaux\\",\\"friendlyName\\":\\"Salon gauche\\",\\"description\\":\\"\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.BrightnessController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"brightness\\",\\"value\\":\\"()=>parseInt('100')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}},{\\"interface\\":\\"Alexa.ColorController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"color\\",\\"value\\":\\"()=>({'hue': undefined,'saturation': undefined, 'brightness': 100})\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('On' === 'Off' || 'On' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"2_YeeLight_Dimmer\\",\\"manufacturerName\\":\\"YeeLight\\",\\"friendlyName\\":\\"Salon milieu\\",\\"description\\":\\"\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.BrightnessController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"brightness\\",\\"value\\":\\"()=>parseInt('100')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}},{\\"interface\\":\\"Alexa.ColorController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"color\\",\\"value\\":\\"()=>({'hue': undefined,'saturation': undefined, 'brightness': 100})\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('On' === 'Off' || 'On' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"3_YeeLight_Dimmer\\",\\"manufacturerName\\":\\"YeeLight\\",\\"friendlyName\\":\\"Salon droite\\",\\"description\\":\\"\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.BrightnessController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"brightness\\",\\"value\\":\\"()=>parseInt('100')\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}},{\\"interface\\":\\"Alexa.ColorController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"color\\",\\"value\\":\\"()=>({'hue': 0,'saturation': 100, 'brightness': 100})\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('On' === 'Off' || 'On' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"5_RFXCom_Blinds\\",\\"manufacturerName\\":\\"RFXCom\\",\\"friendlyName\\":\\"Volet Salon Gauche\\",\\"description\\":\\"06:06:06 2\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"6_RFXCom_Blinds\\",\\"manufacturerName\\":\\"RFXCom\\",\\"friendlyName\\":\\"Volet Cuisine\\",\\"description\\":\\"06:06:06 3\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"4_RFXCom_Blinds\\",\\"manufacturerName\\":\\"RFXCom\\",\\"friendlyName\\":\\"Volet Salon Droite\\",\\"description\\":\\"06:06:06 1\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"7_RFXCom_Blinds\\",\\"manufacturerName\\":\\"RFXCom\\",\\"friendlyName\\":\\"Chambre Emma\\",\\"description\\":\\"06:06:06 4\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"8_RFXCom_Blinds\\",\\"manufacturerName\\":\\"RFXCom\\",\\"friendlyName\\":\\"Chambre Arthur\\",\\"description\\":\\"06:06:06 5\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]},{\\"endpointId\\":\\"9_RFXCom_Blinds\\",\\"manufacturerName\\":\\"RFXCom\\",\\"friendlyName\\":\\"Chambre Parents\\",\\"description\\":\\"06:06:06 6\\",\\"displayCategories\\":[\\"LIGHT\\"],\\"cookie\\":{},\\"capabilities\\":[{\\"type\\":\\"AlexaInterface\\",\\"interface\\":\\"Alexa\\",\\"version\\":\\"3\\"},{\\"interface\\":\\"Alexa.PercentageController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"percentage\\",\\"value\\":\\"0\\"}],\\"retrievable\\":true}},{\\"interface\\":\\"Alexa.PowerController\\",\\"version\\":\\"3\\",\\"type\\":\\"AlexaInterface\\",\\"properties\\":{\\"supported\\":[{\\"name\\":\\"powerState\\",\\"value\\":\\"()=> ('Closed' === 'Off' || 'Closed' === 'Closed') ? 'OFF' : 'ON'\\"}],\\"retrievable\\":true,\\"proactivelyReported\\":true}}]}]}}}"`;
